### ISSUE TITLE
rosenpass-x86_64-darwin-0.2.0 fails to run on macOS Ventura 13.5.1 (22G90)

### ISSUE THREAD

**Issue by TobiasvonCarnap** [rosenpass/rosenpass#120]
rosenpass-x86_64-darwin-0.2.0 on macos ventura 13.5.1 (22G90)
libsodium is installed via homebrew

➜ `sudo /usr/bin/xattr -d com.apple.quarantine rp`
➜ `sudo /usr/bin/xattr -d com.apple.quarantine rosenpass`

➜
```
./rp explain genkey mykey
#! /bin/bash
set -e
umask 077
mkdir -p mykey
wg genkey > mykey/wgsk
rosenpass gen-keys \
  -s mykey/pqsk \
  -p  mykey/pqpk
./rp: Zeile 94: rosenpass: Kommando nicht gefunden.
```

➜
```
./rosenpass keygen \
  private-key mykey/pqsk \
  public-key mykey/pqpk
dyld[56852]: Library not loaded: /nix/store/jbziawf7kv3bvdmn5266vyf2c6cr5j2d-libsodium-1.0.18/lib/libsodium.23.dylib
  Referenced from: <no uuid> /Users/tobi/Downloads/bin/rosenpass
  Reason: tried: '/nix/store/jbziawf7kv3bvdmn5266vyf2c6cr5j2d-libsodium-1.0.18/lib/libsodium.23.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/nix/store/jbziawf7kv3bvdmn5266vyf2c6cr5j2d-libsodium-1.0.18/lib/libsodium.23.dylib' (no such file), '/nix/store/jbziawf7kv3bvdmn5266vyf2c6cr5j2d-libsodium-1.0.18/lib/libsodium.23.dylib' (no such file), '/usr/local/lib/libsodium.23.dylib' (no such file), '/usr/lib/libsodium.23.dylib' (no such file, not in dyld cache)
[1]    56852 abort      ./rosenpass keygen private-key mykey/pqsk public-key mykey/pqpk
```

**Comment by emilengler** [comment]
Can you give me the output of `printenv`? I suspect your LIBRARY_PATH does not honor homebrew.

**Comment by emilengler** [comment]
How are you compiling your rosenpass binary?

**Comment by emilengler** [comment]
@wucke13 Is it possible, that the binaries we provide on the release section are dynamically linked to the crypto libraries inside /nix?

**Comment by wucke13** [comment]
@emilengler yes, that seems to be the case. Unfortunately I do not have a darwin machine handy, so I can not quickly iterate over this. #194 as well suffers from dynamic linking of sodium, but on windows. AFAIK, it is impossible to do static linking on darwin. However, it is possible to link all libraries statically, just the system libraries must be loaded dynamically because the syscall interface is not stable on darwin. This, btw., is the same on windows.

Unfortunately, even automatic static linking of binaries is not an option on Nix for Darwing right now, but there is progress around the corner: https://github.com/NixOS/nixpkgs/issues/214611 & https://github.com/NixOS/nixpkgs/pull/256590 .

I see the following options to get rid of this problem:

- remove all C dependencies (including libsodium)
- wait until the Nix people get mostly-static building for Darwin working
- copy over the `libsodium.so` file to the archive that we release, patch the elf to refer to `./libsodium.so`

Unfortunately, I can not easily iterate on this due to a lack of access to a Darwin machine.

**Comment by wucke13** [comment]
related: #35

**Comment by ilka-schulz** [comment]
- I added this to the #35 tracking issue.
- This is probably related to the CLI re-design (#752, #753, #754).
